base/error.jl

```diff
@@ -50,61 +50,43 @@ macro assert(ex, msgs...)
     :($(esc(ex)) ? $(nothing) : throw(Main.Base.AssertionError($msg)))
 end
 
+const DEFAULT_RETRY_N = 1
+const DEFAULT_RETRY_ON = e->true
+const DEFAULT_RETRY_MAX_DELAY = 10.0
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
+    retry(f, [retry_on]; n=DEFAULT_RETRY_N, max_delay=DEFAULT_RETRY_MAX_DELAY) -> Function
```

Comment:
Agree. Also used the constants in the doc so that any changes are not missed. OTOH, it is still possible to remove/change code and not change the doc string. May as well remove it from the doc string and just have a comment in the code on the need to keep default values and doc strings in sync.